tests/pauli_rotations/circuit_test.py

```diff
@@ -57,6 +65,91 @@ def test_count_rotations_by(circuit, fraction, expected):
+        (([I, X], Fraction(1, 4)), ([Y, X], Fraction(-1, 8))),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Y], Fraction(1, 8))),
+        (([I, Z, I, X], Fraction(-1, 4)), ([Z, X, I, Z], Fraction(-1, 8))),
+    ],
+)
+def test_commute_rotation_to_rotation_commuting(rotation1, rotation2):
+    test_circuit = PauliOpCircuit(len(rotation1[0]))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation1))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation2))
+    test_circuit.commute_pi_over_four_rotation(0)
+
+    assert test_circuit.ops[0].ops_list == rotation2[0]
+    assert test_circuit.ops[0].rotation_amount == rotation2[1]
+    assert test_circuit.ops[1].ops_list == rotation1[0]
+    assert test_circuit.ops[1].rotation_amount == rotation1[1]
+
+
+@pytest.mark.parametrize(
+    "rotation1, rotation2, resultant_rotation",
+    [
+        (([X], Fraction(1, 4)), ([Z], Fraction(1, 4)), ([Y], Fraction(1, 4))),
+        (([Y], Fraction(1, 4)), ([X], Fraction(-1, 8)), ([Z], Fraction(-1, 8))),
+        (([Z, X], Fraction(1, 4)), ([Y, X], Fraction(1, 8)), ([X, I], Fraction(1, 8))),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Z], Fraction(-1, 8)), ([Z, Y, X], Fraction(-1, 8))),
+        (
+            ([X, Y, X, Z, Z], Fraction(1, 4)),
+            ([Z, Y, I, X, Y], Fraction(1, 8)),
+            ([Y, I, X, Y, X], Fraction(1, 8)),
+        ),
+    ],
+)
+def test_commute_rotation_rotation_anti_commuting(rotation1, rotation2, resultant_rotation):
+    test_circuit = PauliOpCircuit(len(rotation1[0]))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation1))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation2))
+    test_circuit.commute_pi_over_four_rotation(0)
+
+    assert test_circuit.ops[0].ops_list == resultant_rotation[0]
+    assert test_circuit.ops[0].rotation_amount == resultant_rotation[1]
+    assert test_circuit.ops[1].ops_list == rotation1[0]
+    assert test_circuit.ops[1].rotation_amount == rotation1[1]
+
+
+@pytest.mark.parametrize(
+    "rotation, measurement",
+    [
+        (([X], Fraction(1, 4)), ([I], False)),
+        (([I, X], Fraction(1, 4)), ([Y, X], True)),
+        (([X, Z, Y], Fraction(1, 4)), ([Y, X, Y], False)),
+        (([I, Z, I, X], Fraction(-1, 4)), ([Z, X, I, Z], True)),
+    ],
+)
+def test_commute_rotation_measurement_commuting(rotation, measurement):
+    test_circuit = PauliOpCircuit(len(rotation[0]))
+    test_circuit.add_pauli_block(PauliRotation.from_list(*rotation))
+    test_circuit.add_pauli_block(Measurement.from_list(*measurement))
+    test_circuit.commute_pi_over_four_rotation(0)
+
+    assert test_circuit.ops[0].ops_list == measurement[0]
+    assert test_circuit.ops[0].isNegative == measurement[1]
+    assert test_circuit.ops[1].ops_list == rotation[0]
+    assert test_circuit.ops[1].rotation_amount == rotation[1]
+
+
+def test_commute_rotation_measurement_anticommute():
+    # TODO: Develop test cases for this test
```

Comment:
good to have Codecov here that is going to keep reminding of the unchecked branch. The proof will be ready soon

Reply:
The same cases from rotation, rotation anticommuting should work here. We have the proof that when they anticoommute the measurement and the Pauli Rotation commute

Reply:
Yes, we do have the proof now